src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -9903,7 +9903,9 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
         ASR::array_physical_typeType physical_type = ASRUtils::extract_physical_type(x_mv_type);
         switch( physical_type ) {
             case ASR::array_physical_typeType::DescriptorArray: {
-                llvm::Value* dim_des_val = arr_descr->get_pointer_to_dimension_descriptor_array(llvm_arg1);
+                llvm::Value* dim_des_val = arr_descr->get_pointer_to_dimension_descriptor_array(
+                    llvm_utils->get_type_from_ttype_t_util(ASRUtils::type_get_past_allocatable(
+                        ASRUtils::type_get_past_pointer(x.m_type)), module.get()), llvm_arg1);                
```

Comment:
I think we need the LLVM type for `x.m_v` here as `x.m_type` refers to the ttype of `ArrayBound`. Could you please try doing the following?

```suggestion
                llvm::Value* dim_des_val = arr_descr->get_pointer_to_dimension_descriptor_array(
                    llvm_utils->get_type_from_ttype_t_util(ASRUtils::type_get_past_allocatable(
                        ASRUtils::type_get_past_pointer(ASRUtils::expr_type(x.m_v))), module.get()), llvm_arg1);                
```

Reply:
Yes it does work. Thanks!